### DISCUSSION TITLE
[QUESTION] Converting a Megatron-LM ckpt to Nemo

### DISCUSSION THREAD

**Discussion by abgoswam** [NVIDIA/Megatron-LM#1251] [Q&A]
Following the [GPT Pretraining](https://github.com/NVIDIA/Megatron-LM/tree/main?tab=readme-ov-file#gpt-pretraining) section in the  `Megatron-LM` repo,  we are able to successfully train a model using  `Megatron-LM` 

I saw pointers on how to convert from HF to `nemo`.  For example, this conversion script  [ convert_llama_hf_to_nemo.py](https://github.com/NVIDIA/NeMo/blob/stable/scripts/checkpoint_converters/convert_llama_hf_to_nemo.py)

However I did not see any examples of converting a ckpt saved using  the  `Megatron-LM` to `nemo` format .  Are there any examples for this?

p.s.  I am thinking of doing this conversion to `nemo`, so I can use tools like `Nemo-Aligner` for post-training

**Comment by gaojingwei** [comment]
I also encountered the same problem in nemo 2.0

**Comment by Phlip79** [comment]
This [doc](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemotoolkit/checkpoints/convert_mlm.html) might be helpful, although it only applies if you are using Nemo 2.5.0 or older.
